app-modules/audit/database/migrations/2024_07_22_092622_data_migration_agent_name.php

```
@@ -0,0 +1,79 @@
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Laravel\Pennant\Feature;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        if (Feature::active('change-agent-name')) {
+            try {
+                DB::beginTransaction();
+
+                $audits = DB::table('audits')
+                    ->join('users', 'audits.change_agent_id', '=', 'users.id')
+                    ->select('audits.id', 'users.name')
+                    ->get();
+
+                foreach ($audits as $audit) {
+                    DB::table('audits')
+                        ->where('id', $audit->id)
+                        ->update(['change_agent_name' => $audit->name]);
+                }
+
+                DB::commit();
+            } catch (Exception $e) {
+                DB::rollBack();
+
+                throw new Exception($e);
```

You can just throw the exception again. You don't need to wrap it in a new one.

In fact, you don't really want to be wrapping it in a new `Exception` as that will cover up what the actual exception class is making it hard to get debugging information.